app/email_batching_test.go

```diff
@@ -232,15 +205,13 @@ func TestCheckPendingNotificationsDefaultInterval(t *testing.T) {
 
 	// notifications should not be sent 1s after post was created, because default batch interval is 15mins
 	job.checkPendingNotifications(time.Unix(10001, 0), func(string, []*batchedNotification) {})
-	if job.pendingNotifications[th.BasicUser.Id] == nil || len(job.pendingNotifications[th.BasicUser.Id]) != 1 {
-		t.Fatal("shouldn't have sent queued post")
-	}
+	require.NotNil(t, job.pendingNotifications[th.BasicUser.Id])
+	require.Len(t, job.pendingNotifications[th.BasicUser.Id], 1, "shouldn't have sent queued post")
 
 	// notifications should be sent 901s after post was created, because default batch interval is 15mins
 	job.checkPendingNotifications(time.Unix(10901, 0), func(string, []*batchedNotification) {})
-	if job.pendingNotifications[th.BasicUser.Id] != nil || len(job.pendingNotifications[th.BasicUser.Id]) != 0 {
-		t.Fatal("should have sent queued post")
-	}
+	require.NotNil(t, job.pendingNotifications[th.BasicUser.Id])
```

Comment:
This require "Nil", not "NotNil". some other similar errors can be found in the jenkins summary here: https://build.mattermost.com/blue/organizations/jenkins/mp%2Fmattermost-server-pr-new/detail/PR-12079/2/tests